PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/serviceexplorer/azure/appservice/SSHTerminalManager.java

```diff
@@ -0,0 +1,203 @@
+            return false;
+        }
+        // check non-docker web app
+        if (StringUtils.containsIgnoreCase(fxVersion, WEB_APP_FX_VERSION_PREFIX)) {
+            DefaultLoader.getUIHelper().showError(SSH_INTO_WEB_APP_ERROR_MESSAGE_NOT_SUPPORT_DOCKER, SSH_INTO_WEB_APP_ERROR_DIALOG_TITLE);
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * create remote connection to remote web app container.
+     *
+     * @param parameters
+     * @return
+     */
+    public CreateRemoteConnectionOutput executeAzCreateRemoteConnectionAndGetOutput(final String[] parameters) {
+        CreateRemoteConnectionOutput connectionInfo = new CreateRemoteConnectionOutput();
+        CommandUtils.CommandExecOutput commendExecOutput = null;
+        try {
+            commendExecOutput = AzureCliUtils.executeCommandAndGetOutputWithCompleteKeyWord(parameters,
+                    AzureCliUtils.CLI_COMMAND_REMOTE_CONNECTION_EXEC_SUCCESS_KEY_WORDS, AzureCliUtils.CLI_COMMAND_REMOTE_CONNECTION_EXEC_FAILED_KEY_WORDS);
+        } catch (IOException | InterruptedException e) {
+            DefaultLoader.getUIHelper().showError(SSH_INTO_WEB_APP_ERROR_MESSAGE, SSH_INTO_WEB_APP_ERROR_DIALOG_TITLE);
+            return connectionInfo;
+        }
+        connectionInfo.setOutputMessage(commendExecOutput.getOutputMessage());
+        connectionInfo.setSuccess(commendExecOutput.isSuccess());
+        if (commendExecOutput.isSuccess()) {
+            String username = PatternUtils.parseWordByPatternAndPrefix(commendExecOutput.getOutputMessage(), PatternUtils.PATTERN_WHOLE_WORD, "username: ");
+            if (StringUtils.isNotBlank(username)) {
+                connectionInfo.setUsername(username);
+            }
+            String port = PatternUtils.parseWordByPatternAndPrefix(commendExecOutput.getOutputMessage(), PatternUtils.PATTERN_WHOLE_NUMBER_PORT, "port: ");
+            if (StringUtils.isNotBlank(port)) {
+                connectionInfo.setPort(port);
+            }
+            String password = PatternUtils.parseWordByPatternAndPrefix(commendExecOutput.getOutputMessage(), PatternUtils.PATTERN_WHOLE_WORD, "password: ");
+            if (StringUtils.isNotBlank(password)) {
+                connectionInfo.setPassword(password);
+            }
+        }
+        return connectionInfo;
+    }
+
+    /**
+     * validate create-remote-connection output to ensure it's ready to ssh to local proxy and open terminal.
+     *
+     * @param connectionInfo
+     * @param webAppName
+     */
+    public boolean validateConnectionOutputBeforeOpenInTerminal(CreateRemoteConnectionOutput connectionInfo, String webAppName) {
+        if (connectionInfo == null) {
+            DefaultLoader.getUIHelper().showError(String.format(SSH_INTO_WEB_APP_ERROR_MESSAGE_WITH_NAME, webAppName), SSH_INTO_WEB_APP_ERROR_DIALOG_TITLE);
```

Comment:
Shall we check all the result parameters and show error dialog once?

Reply:
I would think that checking them with different switch lets the code more readability. and probably, these error message of them maybe different each other one day.